editoast/src/schema/rolling_stock/mod.rs

```diff
@@ -180,6 +180,30 @@ impl<'de> Deserialize<'de> for EffortCurve {
             ));
         }
 
+        let mut last_speeds_value: Option<f64> = None;
+        for i in 0..inner.max_efforts.len() {
+            if inner.max_efforts[i] < 0.0 {
+                return Err(serde::de::Error::custom(
+                    "max_efforts values must be equal or greater than 0.",
+                ));
+            }
+
+            if inner.speeds[i] < 0.0 {
+                return Err(serde::de::Error::custom(
+                    "speeds values must be equal or greater than 0.",
+                ));
+            }
+
+            if let Some(last) = last_speeds_value {
+                if inner.speeds[i] <= last {
+                    return Err(serde::de::Error::custom(
+                        "speeds values must be strictly increasing.",
+                    ));
+                }
+            }
+            last_speeds_value = Some(inner.speeds[i]);
+        }
```

Comment:
A more rusty implementation using the full power of iterators.

```suggestion
        if inner.max_efforts.len() < 2 {
            return Err(serde::de::Error::custom(
                "effort curve should have at least 2 points.",
            ));
        }

        if inner.max_efforts.iter().any(|&x| x < 0.0) {
            return Err(serde::de::Error::custom(
                "max_efforts values must be equal or greater than 0.",
            ));
        };

        if inner.speeds.iter().any(|&x| x < 0.0) {
            return Err(serde::de::Error::custom(
                "speeds values must be equal or greater than 0.",
            ));
        };

        if inner.speeds.windows(2).any(|window| window[0] >= window[1]) {
            return Err(serde::de::Error::custom(
                "speeds values must be strictly increasing.",
            ));
        }
```